staging/src/k8s.io/apimachinery/pkg/runtime/error_test.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtime
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func TestNotRegisteredErr(t *testing.T) {
+	gvk := schema.GroupVersionKind{Group: "test", Version: "v1", Kind: "TestType"}
+	target := GroupVersioner(testGroupVersioner{})
+	schemeName := "testScheme"
+	context := schemeContext("testScheme")
+	testType := reflect.TypeOf(testAPIObject{})
+
+	tests := []struct {
+		name           string
+		err            error
+		expectedErrStr string
+	}{
+		{
+			name:           "with type and target",
+			err:            NewNotRegisteredErrForTarget(schemeName, testType, target),
+			expectedErrStr: fmt.Sprintf("%v is not suitable for converting to %q%s", testType, target, context),
```

Comment:
make `expectedErrStr` the literal string with no formatting so we can see what the final string actually looks like... that would have caught the missing space in schemeContext#String()